specification.markdown

```diff
@@ -12,6 +12,26 @@ EBML uses a simple approach of building Elements upon three pieces of data (tag,
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
 
+## Security Considerations
+
+EBML itself does not offer any kind of security. It has nothing to do with authentication, it dose not provide confidentiality, only marginally useful and effective data integrity options (CRC elements).
+
+EBML does not provide any kind of authorization.
+
+Even if the semantic layer offers any kind of encryption, EBML itself may leak information at either the sematic layer (as declared via the DocType element) or within the EBML structure (you can derive the presence of EBML elements even with an unknown semantic layer with a heuristic approach; not without errors, of course, but with a certain degree of confidence).
+
+Attacks on an EBML reader may include:
+- Invalid Element IDs that are longer than the limit stated in the EBMLMaxIDLength Element of the EBML Header.
+- Invalid Element IDs that not encoded in the shortest-possible way.
+- Invalid Element IDs comprised of reserved values.
+- Invalid Element Data Size values that are longer than the limit stated in the EBMLMaxSizeLength Element of the EBML Header.
```

Comment:
Longer of unsupported, like some float sizes that are not supported or simply don't exist.